components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/ClaimsApiServiceImpl.java

```diff
@@ -30,26 +51,46 @@ public class ClaimsApiServiceImpl extends ClaimsApiService {
     private static final Log LOG = LogFactory.getLog(ClaimsApiServiceImpl.class);
 
     @Override
-    public Response claimsGet(String tenantDomain) {
+    public Response claimsGet(String tenantDomain, String profileName) {
 
-        if (IdentityUtil.threadLocalProperties.get().get(Constants.TENANT_NAME_FROM_CONTEXT) != null) {
-            tenantDomain = (String) IdentityUtil.threadLocalProperties.get().get(Constants.TENANT_NAME_FROM_CONTEXT);
+        if (StringUtils.isBlank(profileName)) {
+            return claimsGet(tenantDomain);
         }
 
-        if (StringUtils.isBlank(tenantDomain)) {
-            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
-        } else if (!RecoveryUtil.isValidTenantDomain(tenantDomain)) {
-            RecoveryUtil.handleBadRequest("Invalid tenant domain :" + tenantDomain, IdentityRecoveryConstants
-                    .ErrorMessages.ERROR_CODE_INVALID_TENANT.getCode());
+        tenantDomain = resolveTenantDomain(tenantDomain);
+        ClaimDTO[] claimDTOs = new ClaimDTO[0];
+        try {
+            List<LocalClaim> localClaims = RecoveryUtil.getClaimMetadataManagementService()
+                    .getSupportedLocalClaimsForProfile(tenantDomain, profileName);
+            if (CollectionUtils.isEmpty(localClaims)) {
+                claimDTOs = new ClaimDTO[0];
+            }
+            claimDTOs = RecoveryUtil.getClaimDTOs(localClaims);
+        } catch (ClaimMetadataException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Client Error while getting all identity claims ", e);
+            }
+            RecoveryUtil.handleBadRequest(e.getMessage(), e.getErrorCode());
+        } catch (Throwable throwable) {
+            RecoveryUtil.handleInternalServerError(Constants.SERVER_ERROR, IdentityRecoveryConstants
+                    .ErrorMessages.ERROR_CODE_UNEXPECTED.getCode(), LOG, throwable);
         }
+        return Response.ok(claimDTOs).build();
+    }
+
+    @Override
+    public Response claimsGet(String tenantDomain) {
+
+        tenantDomain = resolveTenantDomain(tenantDomain);
```

Comment:
Why we can't use this method.
IdentityTenantUtil.getTenantDomainFromContext())

Reply:
Din't check the tenant domain resolving logic here,  just reused the old code for that and moved that to a method. In this case tenant domain is passed as a query param to this API as well.